lib/src/rules/use_build_context_synchronously.dart

```diff
@@ -17,11 +17,13 @@ const _details = r'''
 **DON'T** use BuildContext across asynchronous gaps.
 
 Storing `BuildContext` for later usage can easily lead to difficult to diagnose
-crashes. Asynchronous gaps are implicitly storing `BuildContext` and are some of
-the easiest to overlook when writing code.
+crashes, because it can change its state during the async gap and no longer be 
+asociated to an `Element`. Asynchronous gaps are implicitly storing 
+`BuildContext` and are some of the easiest to overlook when writing code.
 
-When a `BuildContext` is used, its `mounted` property must be checked after an
-asynchronous gap.
```

Comment:
Please keep this advice. It is the number one solution for fixing this lint. 

(Like I said in another comment, it only applies to newer versions of flutter, older versions only provide the `mounted` property on the `State` and it should be checked there instead if one uses an older version of flutter.)